x/incentive/keeper/keeper_fees_test.go

```diff
@@ -94,3 +94,139 @@ func TestCollectGasFeesToIncentiveModule(t *testing.T) {
+	argExitFee := sdk.MustNewDecFromStr("0.1")
+
+	poolParams := &ammtypes.PoolParams{
+		SwapFee: argSwapFee,
+		ExitFee: argExitFee,
+	}
+
+	msg := types.NewMsgCreatePool(
+		addr[0].String(),
+		poolParams,
+		poolAssets,
+	)
+
+	// Create a Elys+USDC pool
+	poolId, err := amm.CreatePool(ctx, msg)
+	require.NoError(t, err)
+	require.Equal(t, poolId, uint64(0))
+
+	// ####### POOL 2 ########
+	// ATOM+USDC pool
+	// Mint uusdc
+	usdcToken = sdk.NewCoins(sdk.NewCoin(ptypes.USDC, sdk.NewInt(200000)))
+
+	err = app.BankKeeper.MintCoins(ctx, types.ModuleName, usdcToken)
+	require.NoError(t, err)
+	err = app.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, addr[1], usdcToken)
+	require.NoError(t, err)
+
+	// Mint uatom
+	atomToken := sdk.NewCoins(sdk.NewCoin(ptypes.ATOM, sdk.NewInt(200000)))
+	err = bk.MintCoins(ctx, types.ModuleName, atomToken)
+	require.NoError(t, err)
+	err = bk.SendCoinsFromModuleToAccount(ctx, types.ModuleName, addr[1], atomToken)
+	require.NoError(t, err)
+
+	var poolAssets2 []ammtypes.PoolAsset
+	// Atom
+	poolAssets2 = append(poolAssets2, ammtypes.PoolAsset{
+		Weight: sdk.NewInt(50),
+		Token:  sdk.NewCoin(ptypes.ATOM, sdk.NewInt(150000)),
+	})
+
+	// USDC
+	poolAssets2 = append(poolAssets2, ammtypes.PoolAsset{
+		Weight: sdk.NewInt(50),
+		Token:  sdk.NewCoin(ptypes.USDC, sdk.NewInt(10000)),
+	})
+
+	msg = types.NewMsgCreatePool(
+		addr[1].String(),
+		poolParams,
+		poolAssets2,
+	)
+
+	// Create a ATOM+USDC pool
+	poolId, err = amm.CreatePool(ctx, msg)
+	require.NoError(t, err)
+	require.Equal(t, poolId, uint64(1))
+
+	pools := amm.GetAllPool(ctx)
+
+	// check length of pools
+	require.Equal(t, len(pools), 2)
+
+	// check block height
+	require.Equal(t, int64(0), ctx.BlockHeight())
+
+	// Fill in pool #1 revenue wallet
+	revenueAddress1 := ammtypes.NewPoolRevenueAddress(0)
+	usdcRevToken1 := sdk.NewCoins(sdk.NewCoin(ptypes.USDC, sdk.NewInt(1000)))
+	err = app.BankKeeper.MintCoins(ctx, types.ModuleName, usdcRevToken1)
+	require.NoError(t, err)
+	err = app.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, revenueAddress1, usdcRevToken1)
+	require.NoError(t, err)
+
+	// Fill in pool #2 revenue wallet
+	revenueAddress2 := ammtypes.NewPoolRevenueAddress(1)
+	usdcRevToken2 := sdk.NewCoins(sdk.NewCoin(ptypes.USDC, sdk.NewInt(2000)))
+	err = app.BankKeeper.MintCoins(ctx, types.ModuleName, usdcRevToken2)
+	require.NoError(t, err)
+	err = app.BankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, revenueAddress2, usdcRevToken2)
+	require.NoError(t, err)
+
+	// Collect revenue
+	collectedAmt, rewardForLpsAmt := ik.CollectDEXRevenue(ctx)
+
+	// check block height
+	require.Equal(t, int64(0), ctx.BlockHeight())
+
+	// It should be 3000=1000+2000 usdc
+	require.Equal(t, collectedAmt, sdk.Coins{sdk.NewCoin(ptypes.USDC, sdk.NewInt(3000))})
+	// It should be 1950=3000*0.65 usdc
+	require.Equal(t, rewardForLpsAmt, sdk.DecCoins{sdk.NewDecCoin(ptypes.USDC, sdk.NewInt(1950))})
```

Comment:
It would be nice to add more test cases using table-driven approach.

Reply:
It will be thousands of lines for the testing then as the preparing is huge. :)

Reply:
once we add some code coverage reporting we will be able to know what need to be covered by tests